### PR TITLE
Homework/week1

### PR DESCRIPTION
**Pull request by HongDiep1806**
i have just finished a few exercises, let take a look and give me comments
I implemented this ticket https://trello.com/c/etOcJFA5/4-food-searching

### CODE REVIEW

File: Add to Array-Form of Integer/Add to Array-Form of Integer/Program.cs [hunk 1]
```diff
@@ -0,0 +1,65 @@
+﻿namespace Add_to_Array_Form_of_Integer
+{
+    internal class Program
+    {
+        static void Main(string[] args)
+        {
+            Reader reader = new Reader();
```
Comment (linlekms) [inline Add to Array-Form of Integer/Add to Array-Form of Integer/Program.cs:7]: You can use var to declare variables

File: Add to Array-Form of Integer/Add to Array-Form of Integer/Program.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            var array = new int[] { 1, 3, 0, 1 };
+            var k = 34;
+            var number = 0;
+            number = getNumber(array);
```
Comment (linlekms) [inline Add to Array-Form of Integer/Add to Array-Form of Integer/Program.cs:11]: You can shorten these 2 lines.

Comment (linlekms) [reply Add to Array-Form of Integer/Add to Array-Form of Integer/Program.cs:11]:
```suggestion
            var number =  getNumber(array);
```

File: Add to Array-Form of Integer/Add to Array-Form of Integer/Program.cs [hunk 3]
```diff
@@ -0,0 +1,65 @@
+﻿namespace Add_to_Array_Form_of_Integer
```
Comment (linlekms) [inline Add to Array-Form of Integer/Add to Array-Form of Integer/Program.cs:1]: I cannot find this problem in the coding problems sheet, please send me the URL.

File: Add to Array-Form of Integer/Add to Array-Form of Integer/Program.cs [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        static void Main(string[] args)
+        {
+            Reader reader = new Reader();
+            var array = new int[] { 1, 3, 0, 1 };
+            var k = 34;
+            var number = 0;
+            number = getNumber(array);
+            Console.WriteLine(number+k);
+        }
+        public static int getNumber(int[] array)
```
Comment (linlekms) [inline Add to Array-Form of Integer/Add to Array-Form of Integer/Program.cs:14]: Add a line to separate these 2 functions

File: Boy or Girl/Boy or Girl/Program.cs [hunk 5]
```diff
@@ -0,0 +1,51 @@
+﻿namespace Boy_or_Girl
+{
+    internal class Program
+    {
+        static void Main(string[] args)
+        {
+            Reader reader = new Reader();
+            var userName = reader.Next();
+            var check = checkGirl(userName);
+            if (check == true)
```
Comment (linlekms) [inline Boy or Girl/Boy or Girl/Program.cs:10]: Check is a boolean value itself so we donot need to compare it with another boolean value.

Comment (linlekms) [reply Boy or Girl/Boy or Girl/Program.cs:10]:
```suggestion
            var check = checkGirl(userName);
            if (check)
```

Comment (linlekms) [reply Boy or Girl/Boy or Girl/Program.cs:10]: Or we can shorten it as following suggestion:
```suggestion
            if (checkGirl(userName))
```

File: Boy or Girl/Boy or Girl/Program.cs [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+                Console.WriteLine("IGNORE HIM!");
+            }
+        }
+        public static Boolean checkGirl(String userName)
+        {
+            if ((userName.Distinct().Count() % 2) != 0)
+            {
+                return false;
+            }
+            return true;
```
Comment (linlekms) [inline Boy or Girl/Boy or Girl/Program.cs:25]: We can shorten this function as following suggestion: 
```suggestion
            return !((userName.Distinct().Count() % 2) != 0);
```

File: Move Zeroes/Move Zeroes/Program.cs [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+            }
+            for (int i = 0; i < count; i++)
+            {
+                arrayList.Add(0);
+            }
+            var arrayResult = arrayList.ToArray();
+            for (int i = 0; i < arrayResult.Length; i++)
+            {
+                Console.Write(arrayResult[i] + " ");
+            }
```
Comment (linlekms) [inline Move Zeroes/Move Zeroes/Program.cs:32]: Can we remove the arrayResult and just use arrayList variable here?

File: Next Round/Next Round/Program.cs [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+            for (int i = 0; i < numOfContestant; i++)
+            {
+                scores[i] = reader.NextInt();
+            }
+
+            var result = count(scores, k);
+            Console.WriteLine(result);
+
+        }
+        class Reader
```
Comment (linlekms) [inline Next Round/Next Round/Program.cs:21]: Add a line to separate these 2 functions

File: Team/Team/Program.cs [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+
+        public static Boolean Check(int num)
+        {
+            Boolean check = true;
+            if (num == 0)
+            {
+                return check = false;
+            }
+            return check;
+        }
```
Comment (linlekms) [inline Team/Team/Program.cs:41]:
```suggestion
        public static Boolean Check(int num)
        {
            return (num != 0);
        }
```

File: Team/Team/Program.cs [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+            var count = 0;
+            while (numOfQuestion > 0)
+            {
+                var solutions = new int[3];
+                var countIn = 0;
+                for (int i = 0; i < 3; i++)
+                {
+                    solutions[i] = reader.NextInt();
+                    Boolean check = Check(solutions[i]);
+                    if (check == true)
```
Comment (linlekms) [inline Team/Team/Program.cs:18]:
```suggestion
                    if ( Check(solutions[i]))
```

**Review by linlekms — COMMENTED** [review]
Hi @HongDiep1806 ,
I finished my first review round, please take a look and improve your code.

File: Add to Array-Form of Integer/Add to Array-Form of Integer/Program.cs (hunk 3 above)
Comment (HongDiep1806) [reply Add to Array-Form of Integer/Add to Array-Form of Integer/Program.cs:1]: That's : Leetcode 989
https://leetcode.com/problems/add-to-array-form-of-integer/description/

**Pull request merged**